packages/@aws-cdk/cloud-assembly-schema/lib/manifest.ts

```diff
@@ -99,7 +99,7 @@ export class Manifest {
     if (semver.gt(actual, maxSupported)) {
       // we use a well known error prefix so that the CLI can identify this specific error
       // and print some more context to the user.
-      throw new Error(`${VERSION_MISMATCH}: Maximum schema version supported is ${maxSupported}, but found ${actual}`);
+      throw new Error(`${VERSION_MISMATCH}: The CLI scheme version found is ${maxSupported}, but the framework version used requires at least ${actual}`);
```

Comment:
Per @iliapolo suggestion, I agree we should remove this text an simply leave:

>This CDK CLI is not compatible with the CDK library used by your application. Please upgrade the CLI to the latest version.

If we do that, it seems like we can simply throw a custom error. @RomainMuller any concerns? Since the cx-api is not a jsii module it shouldn't be a problem

Reply:
No problem here... Outside of the jsii context this is no problemo.

In a jsii context it'd be annoying - if you intend on a control flow where users can rescue from *this specific* exception, then you'd end up disappointed because it would surface as an undifferentiated Exception/Error class in other languages, defeating the purpose.

Reply:
Dammit! So the cx-api is not a jsii module, however the cloudAssembly module is (not sure why that is actually). Since the custom error type is thrown from the CloudAssembly module and used in another module, it needs to be exported:

```ts
export class VersionMismatchError extends Error { 
...
}
```

Which jsii does not like. 

@RomainMuller I'm just wondering, is there a way do this? Given that the custom error type is only used as control flow in the cx-api module?